packages/smooth_app/lib/pages/product/common/product_list_helper.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_refresher.dart';
+
+// ignore: avoid_classes_with_only_static_members
+class ProductListHelper {
+  static const String EXPORT_SEPARATOR = ';';
+
+  static Future<String> exportList(
+    ProductList list,
+    LocalDatabase localDatabase,
+  ) async {
+    final StringBuffer csv = StringBuffer();
+
+    void addLine(String barcode, String name, String brand) {
+      csv.writeln(
+        '$barcode'
+        '$EXPORT_SEPARATOR'
+        '$name'
+        '$EXPORT_SEPARATOR'
+        '$brand',
+      );
+    }
+
+    addLine('Barcode', 'Name', 'Brand');
+
+    for (final String barcode in list.barcodes) {
+      if (barcode.length != 8 ||
+          barcode.length != 13 ||
+          int.tryParse(barcode) == null) {
+        continue;
+      }
+
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name =
+            (product.productName ?? '').replaceAll(EXPORT_SEPARATOR, '');
+        final String brand =
+            (product.brands ?? '').replaceAll(EXPORT_SEPARATOR, '');
+        addLine(barcode, name, brand);
+      }
+    }
+
+    return csv.toString();
+  }
+
+  static Future<void> importList(
```

Comment:
I would return `List<String> barcodes`, so we can check how many barcodes were eventually loaded.

I would even split in two very distinct methods:
* the first splitting the file into a `List<String>` of barcodes
* the second loading a `List<String>` of barcodes